instrumentation/lettuce/lettuce-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/lettuce/v4_0/LettuceSyncClientTest.java

```diff
@@ -0,0 +1,307 @@
+  }
+
+  @Test
+  void testListCommand() {
+    long res = syncCommands.lpush("TESTLIST", "TESTLIST ELEMENT");
+    assertThat(res).isEqualTo(1);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("LPUSH")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "LPUSH"))));
+  }
+
+  @Test
+  void testHashSetCommand() {
+    String res = syncCommands.hmset("user", testHashMap);
+    assertThat(res).isEqualTo("OK");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("HMSET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "HMSET"))));
+  }
+
+  @Test
+  void testHashGetallCommand() {
+    Map<String, String> res = syncCommands.hgetall("TESTHM");
+    assertThat(res).isEqualTo(testHashMap);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("HGETALL")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "HGETALL"))));
+  }
+
+  @Test
+  void testDebugSegfaultCommandWithNoArgumentShouldProduceSpan() {
+    // Test Causes redis to crash therefore it needs its own container
+    GenericContainer<?> server =
+        new GenericContainer<>(DockerImageName.parse("redis:6.2.3-alpine")).withExposedPorts(6379);
+    server.start();
+
+    long serverPort = server.getMappedPort(6379);
+    RedisClient client = RedisClient.create("redis://" + host + ":" + serverPort + "/" + DB_INDEX);
+    StatefulRedisConnection<String, String> connection1 = client.connect();
+    RedisCommands<String, String> commands = connection1.sync();
+    // 1 connect trace
+    testing.clearData();
+
+    commands.debugSegfault();
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("DEBUG")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "redis"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "DEBUG"))));
+
+    // Server already crashed but just in case
+    connection1.close();
+    server.stop();
```

Comment:
Nit: you can use the [`AutoCleanupExtension`](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/testing-common/src/main/java/io/opentelemetry/instrumentation/testing/internal/AutoCleanupExtension.java) to ensure that these will 100% get closed at the end of test

Reply:
thanks for this suggestion!